dcpy/lifecycle/builds/plan.py

```diff
@@ -55,6 +55,25 @@ def plan_recipe(recipe_path: Path, version: str | None = None) -> Recipe:
                     bump_type=recipe.version_type,
                     bump_by=bump.bump_latest_release,
                 ).label
+            case versions.PinToSourceDataset():
+                dataset = recipe.version_strategy.pin_to_source_dataset
+                inputs_by_name = {d.name: d for d in recipe.inputs.datasets}
+                if dataset not in inputs_by_name:
+                    raise ValueError(
+                        f"Cannot pin build version to dataset '{dataset}' as it is not an input dataset"
+                    )
+                input = inputs_by_name[dataset]
+                if not input.version and (
+                    input.version_env_var
```

Comment:
could you please explain the line above? From my understanding, if `input.version` is not specified AND env var is present, then you get an error...

Reply:
Yeah sorry, it definitely is not intuitive to read.

Basically, I decided that it made sense to only allow pinning to this version if either
1. version is explicitly defined explicitly (great - it's used)
2. version is not defined explicitly and "latest" is being used

All other cases of not defining it explicitly should error - so either no explicit version, but `version_env_var ` being defined, or no explicit version and the missing version strategy is something other than "find_latest"

Reply:
The only env var we currently use is `VERSION_PREV`, which makes no sense to pin to. In general, thought it made sense to just limit this a little - only allow explicit definition or "latest", and be overly cautious throwing errors

Reply:
Got it, thank you